axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/SaleOrderInvoiceServiceImpl.java

```diff
@@ -612,8 +612,9 @@ public Invoice generate() throws AxelorException {
 
       if (qtyToInvoiceMap.containsKey(saleOrderLine.getId())) {
         List<InvoiceLine> invoiceLines =
-            this.createInvoiceLine(
-                invoice, saleOrderLine, qtyToInvoiceMap.get(saleOrderLine.getId()));
+            Beans.get(SaleOrderInvoiceService.class)
```

Comment:
Please do not inject a service from itself, you do not have to inject again the service to get overridden methods